src/functions/InModuleScope.ps1

```diff
@@ -75,26 +75,48 @@
     )
 
     $module = Get-ScriptModule -ModuleName $ModuleName -ErrorAction Stop
+    $sessionState = Set-SessionStateHint -PassThru -Hint "Module - $($module.Name)" -SessionState $module.SessionState
 
-    # TODO: could this simply be $PSCmdlet.SessionState? Because the original scope we are moving from
-    # is the scope in which this command is running, right?
-    # $originalState = $Pester.SessionState
-    # $originalScriptBlockScope = Get-ScriptBlockScope -ScriptBlock $ScriptBlock
-
-    # try {
-    # $sessionState = Set-SessionStateHint -PassThru -Hint "Module - $($module.Name)" -SessionState $module.SessionState
-    # $Pester.SessionState = $sessionState
-    # Set-ScriptBlockScope -ScriptBlock $ScriptBlock -SessionState $sessionState
-
-    # do {
-    # Write-ScriptBlockInvocationHint -Hint "InModuleScope" -ScriptBlock $ScriptBlock
-    & $module $ScriptBlock @Parameters @ArgumentList
-    # } until ($true)
-    # }
-    # finally {
-    # $Pester.SessionState = $originalState
-    # Set-ScriptBlockScope -ScriptBlock $ScriptBlock -SessionStateInternal $originalScriptBlockScope
-    # }
+    $wrapper = {
+        param (
+            [Parameter(Mandatory = $true)]
+            [scriptblock]
+            ${Script Block},
```

Comment:
Consider using $private: so the variables don't leak to child scopes. In here you don't have any variables that you need to use in child scope, so it's a better way than having all of those variables have weird names. But I see that you are just copying the same approach from a different place in the code base.

Reply:
See below for a link where you can mostly just copy paste the implementation.

Reply:
Yeah, I stripped down behavior execution. 👍 Will absolutely simplify and use private-scope. Thanks